data/transactions/logic/eval.go

```diff
@@ -1506,7 +1494,7 @@ func opBalance(cx *evalContext) {
 		return
 	}
 
-	addr, err := getAccountAddrByOffset(cx.Txn, accountIdx)
+	addr, err := cx.Txn.Txn.AddressByIndex(accountIdx, cx.Txn.Txn.Sender)
```

Comment:
In theory `AddressByIndex` might be internal to appLedger and all appLedger methods might accept account indexes. Mb in another PR